### PR TITLE
HDDS-10309. Speed up TestSnapshotDeletingService

### PR DESCRIPTION
**Pull request by raju-balpande**
## What changes were proposed in this pull request?
Speed up TestSnapshotDeletingService
The time span is improved to 250.118 seconds from 321.147 seconds earlier.
https://github.com/raju-balpande/apache_ozone/actions/runs/8339044173/job/22820842037 

Please describe your PR in detail:
We have managed the cluster creation part to be executed once.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10309?

## How was this patch tested?
Tested each method independently and in sequence locally and Tested the sequence in Ci https://github.com/raju-balpande/apache_ozone/actions/runs/8339044173/job/22820842037

### CODE REVIEW

**Review by hemantk-12 — COMMENTED** [review]
Thanks @raju-balpande for the patch.

Is it possible to create a separate bucket for each of the tests? Rather than using the order of tests.

@aswinshakil please take a look as well.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestSnapshotDeletingService.java [hunk 1]
```diff
@@ -189,10 +206,16 @@ public void testMultipleSnapshotKeyReclaim() throws Exception {
     // verify the cache of purged snapshot
     // /vol1/bucket2/bucket2snap1 has been cleaned up from cache map
     assertEquals(2, om.getOmSnapshotManager().getSnapshotCacheSize());
+    //clean data
+    //    // cleaning up the data
```
Comment (hemantk-12) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestSnapshotDeletingService.java:210]: nit: can you please fix the duplicate comments?

**Review by hemantk-12 — COMMENTED** [review]
Overall looks good to me. Left some minor comments.

Can we avoid order if we create a new bucket for each test?

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestSnapshotDeletingService.java [hunk 2]
```diff
@@ -81,8 +90,11 @@ public class TestSnapshotDeletingService {
   private static final String VOLUME_NAME = "vol1";
   private static final String BUCKET_NAME_ONE = "bucket1";
   private static final String BUCKET_NAME_TWO = "bucket2";
+  private static final String BUCKET_NAME_FSO = "bucketfso";
+
+  private boolean runIndividualTest = true;
```
[the remark this answers is not in the document]
Comment (raju-balpande) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestSnapshotDeletingService.java:95]: By this check, I am making sure that the If anyone just want to run a single method it shall work for him.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestSnapshotDeletingService.java (hunk 1 above)
Comment (raju-balpande) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestSnapshotDeletingService.java]: Done. thanks.

**Pull request closed without merging**